src/coreclr/clrdefinitions.cmake

```diff
@@ -226,7 +226,7 @@ if(CLR_CMAKE_TARGET_WIN32)
   endif(CLR_CMAKE_TARGET_ARCH_AMD64 OR CLR_CMAKE_TARGET_ARCH_I386)
 endif(CLR_CMAKE_TARGET_WIN32)
 
-if(CLR_CMAKE_TARGET_OSX)
+if(CLR_CMAKE_TARGET_OSX AND NOT CLR_CMAKE_TARGET_ARCH_ARM64)
   add_definitions(-DFEATURE_WRITEBARRIER_COPY)
```

Comment:
It is not clear to me how important this feature is.  The assembly files are missing on arm64.  If it is important I could port them from amd64.

Reply:
This was introduced by dotnet/coreclr#26193.  Looks like it was part of Mojave hardened runtime support.  

```
Move JIT_WriteBarrier that is modified at runtime to a dynamically  allocated memory 
instead of making a page in libcoreclr.dylib RWX
```

Since we have to fix W^X globally, it is still not clear we will need this.

Reply:
This is needed for performance optimization that patches GC write barriers to avoid unnecessary indirections. 

This optimization on ARM64 is implemented by `JIT_UpdateWriteBarrierState` and friends.